agents/rust/mediator/tests/mediator-protocol-pickup.rs

```diff
@@ -0,0 +1,263 @@
+        pickup::{
+            DeliveryRequest, DeliveryRequestContent, DeliveryRequestDecorators, Pickup,
+            StatusRequest, StatusRequestContent, StatusRequestDecorators,
+        },
+    },
+    AriesMessage,
+};
+
+use crate::common::{
+    agent_and_transport_utils::{
+        gen_mediator_connected_agent, send_message_and_pop_response_message,
+    },
+    prelude::*,
+    test_setup::setup_env_logging,
+};
+
+static LOGGING_INIT: std::sync::Once = std::sync::Once::new();
+
+async fn get_mediator_grant_data(
+    agent: &Agent<impl BaseWallet + 'static, impl MediatorPersistence>,
+    agent_aries_transport: &mut impl AriesTransport,
+    agent_verkey: &VerKey,
+    mediator_diddoc: &AriesDidDoc,
+) -> MediateGrantData {
+    // prepare request message
+    let message = MediatorCoordMsgEnum::MediateRequest;
+    let message_bytes = serde_json::to_vec(&message).unwrap();
+    // send message and get response
+    let response_message = send_message_and_pop_response_message(
+        &message_bytes,
+        agent,
+        agent_aries_transport,
+        agent_verkey,
+        mediator_diddoc,
+    )
+    .await
+    .unwrap();
+    // extract routing parameters
+    if let MediatorCoordMsgEnum::MediateGrant(grant_data) =
+        serde_json::from_str(&response_message).unwrap()
+    {
+        info!("Grant Data {:?}", grant_data);
+        grant_data
+    } else {
+        panic!(
+            "Should get response that is of type Mediator Grant. Found {:?}",
+            response_message
+        )
+    }
+}
+
+/// Register recipient keys with mediator
+async fn gen_and_register_recipient_key(
+    agent: &mut Agent<impl BaseWallet + 'static, impl MediatorPersistence>,
+    agent_aries_transport: &mut impl AriesTransport,
+    agent_verkey: &VerKey,
+    mediator_diddoc: &AriesDidDoc,
+) -> Result<(VerKey, AriesDidDoc)> {
+    let agent_invite: OOBInvitation = agent
+        .get_oob_invite()
+        .map_err(|e| GenericStringError { msg: e.to_string() })?;
+    let agent_diddoc = oob2did(agent_invite);
+    let agent_recipient_key = agent_diddoc
+        .recipient_keys()
+        .unwrap()
+        .first()
+        .unwrap()
+        .clone();
+    // register recipient key with mediator
+    let message = MediatorCoordMsgEnum::KeylistUpdateRequest(KeylistUpdateRequestData {
+        updates: vec![KeylistUpdateItem {
+            recipient_key: agent_recipient_key.clone(),
+            action: KeylistUpdateItemAction::Add,
+            result: None,
+        }],
+    });
+    info!("Sending {:?}", serde_json::to_string(&message).unwrap());
+    let message_bytes = serde_json::to_vec(&message)?;
+    let _response_message = send_message_and_pop_response_message(
+        &message_bytes,
+        agent,
+        agent_aries_transport,
+        agent_verkey,
+        mediator_diddoc,
+    )
+    .await?;
+    Ok((agent_recipient_key, agent_diddoc))
+}
+
+async fn forward_dummy_anoncrypt_message(
```

Comment:
dummy?

Reply:
It's simply a Basic Message with a trivial hello string wrapped in an encrypted envelope, used for the test.
 
Thanks for the catch, I shall rename to `forward_basic_anoncrypt_message` as this is could cause confusion that it isn't a real message (it is!).

Reply:
Ah, I've missed the fact this is part of testing